grommet/grommet
# [TS] Migrate collapsible tests to tsx

#### What does this PR do?

This PR migrates Collapsible component tests from `.js` to `.tsx`.

#### Where should the reviewer start?

- Make sure all tests pass
- Review background context section below to see what was changed and why.

#### What testing has been done on this PR?

- Ran tests manually and passed all tests locally
- Awaiting CircleCI Test results...

#### How should this be manually tested?

- The reviewer can review and verify that the tests run properly.
-  The reviewer can see comments in background context to see reasoning behind certain changes made in this PR.

#### Any background context you want to provide?

Upon switch to `.tsx` and initial run of `ts-jest` I got the following output:

Input: `yarn test Collapsible`
Output: 
```
yarn run v1.22.10
warning ../package.json: No license field
$ jest --runInBand Collapsible
 FAIL  src/js/components/Collapsible/__tests__/Collapsible-test.tsx
  ● Test suite failed to run

    src/js/components/Collapsible/__tests__/Collapsible-test.tsx:13:7 - error TS7034: Variable 'App' implicitly has type 'any' in some locations where its type cannot be determined.

    13   let App;
             ~~~
    src/js/components/Collapsible/__tests__/Collapsible-test.tsx:27:35 - error TS7005: Variable 'App' implicitly has an 'any' type.

    27     const { container } = render(<App />);
                                         ~~~
    src/js/components/Collapsible/__tests__/Collapsible-test.tsx:33:35 - error TS7005: Variable 'App' implicitly has an 'any' type.

    33     const { container } = render(<App open />);
                                         ~~~
    src/js/components/Collapsible/__tests__/Collapsible-test.tsx:38:45 - error TS7005: Variable 'App' implicitly has an 'any' type.

    38     const { container, rerender } = render(<App open />);
                                                   ~~~
    src/js/components/Collapsible/__tests__/Collapsible-test.tsx:40:15 - error TS7005: Variable 'App' implicitly has an 'any' type.

    40     rerender(<App open={false} />);
                     ~~~

Jest: "global" coverage threshold for statements (86.9%) not met: 0%
Jest: "global" coverage threshold for branches (76.3%) not met: 0%
Jest: "global" coverage threshold for lines (87.9%) not met: 0%
Jest: "global" coverage threshold for functions (88%) not met: 0%
Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        30.741 s
Ran all test suites matching /Collapsible/i.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

- The main issue here is that this test is predefining an example `App` before each render which can be passed a list of props. Although we can give the `App` variable declaration a defined type, I feel that it is a bit safer to call each render with the resulting jsx explicitly declared. This will fix the error as well as make things clearer to those reviewing the tests.

- Looking at the PR that introduced the `beforeEach` call (https://github.com/grommet/grommet/pull/4141#issuecomment-642222706), it seems that it was introduced to remove redundant lines, however I do feel my point that it isn't necessarily clear still stands.

#### What are the relevant issues?

#5599 

#### Do the grommet docs need to be updated?

No

#### Should this PR be mentioned in the release notes?

No

#### Is this change backwards compatible or is it a breaking change?

Backwards compatible
